java/code/src/com/redhat/rhn/frontend/servlets/AuthorizationFilter.java

```diff
@@ -0,0 +1,198 @@
+                        " is not available for unauthenticated users.");
+            }
+        }
+        else {
+            if (!user.hasRole(RoleFactory.SAT_ADMIN)) {
+                Optional<WebEndpoint> endpointOpts = WebEndpointFactory.lookupByUserIdEndpointScope(user.getId(),
+                        path,
+                        hreq.getMethod(),
+                        WebEndpoint.Scope.W);
+                if (endpointOpts.isEmpty()) {
+                    throw new PermissionException("The " + hreq.getRequestURI() +
+                            " API is not available to user " + user.getLogin());
+                }
+            }
+        }
+    }
+
+    private void handleStrutsAccess(HttpServletRequest hreq, Set<String> noAuthEndpoints) {
+        // Check if the request URI needs authentication. If so, also check if the URI needs authorization as well.
+        if (!authenticationService.requestURIRequiresAuthentication(hreq) ||
+                noAuthEndpoints.contains(hreq.getServletPath())) {
+            return;
+        }
+
+        User user = new RequestContext(hreq).getCurrentUser();
+        if (user == null) {
+            if (!noAuthEndpoints.contains(hreq.getServletPath())) {
+                throw new PermissionException("The " + hreq.getRequestURI() +
+                        " is not available for unauthenticated users.");
+            }
+        }
+        else {
+            if (!user.hasRole(RoleFactory.SAT_ADMIN)) {
+                Optional<WebEndpoint> endpointOpts = WebEndpointFactory.lookupByUserIdEndpointScope(user.getId(),
+                        hreq.getServletPath(),
+                        hreq.getMethod(),
+                        WebEndpoint.Scope.W);
+                if (endpointOpts.isEmpty()) {
+                    throw new PermissionException("The " + hreq.getRequestURI() +
+                            " API is not available to user " + user.getLogin());
+                }
+            }
+        }
+    }
+
+    private void handleSparkAccess(HttpServletRequest hreq, Set<String> noAuthEndpoints) {
+        RouteMatch route = ServletRoutes.get().find(
+                HttpMethod.get(hreq.getMethod().toLowerCase()),
+                hreq.getServletPath(), hreq.getContentType());
+        if (route == null)  {
+            // TODO: This needs to be 404
+            throw new PermissionException("Route not found to verify authorization for: " + hreq.getRequestURI());
+        }
+
+        // Check if the request URI needs authentication. If so, also check if the URI needs authorization as well.
+        if (!authenticationService.requestURIRequiresAuthentication(hreq) ||
+                noAuthEndpoints.contains(route.getMatchUri())) {
+            return;
+        }
+
+        RequestContext requestContext = new RequestContext(hreq);
+        User user = requestContext.getCurrentUser();
+        if (user == null) {
+            if (!noAuthEndpoints.contains(route.getMatchUri())) {
```

Comment:
I think we can just throw the exception without the if statement, since we are already checking the `noAuthEndpoints` in line 173.
HOwever, it doesn't cause any issue if we keep it as is. Up to you to decide :)

Reply:
Right, I guess I overlooked this one.